css: Improve the css of icon-based tab picker used in topic popover.

prakhar1144 opened the pull request:
[CZO Discussion
](https://chat.zulip.org/#narrow/stream/9-issues/topic/Topic.20mute.2Fdefault.2Ffollow.20buttons.20have.20odd.20hover.20effect/near/1658106)

| Before | After |
|--------|--------|
| ![Screenshot from 2023-10-09 00-54-03](https://github.com/zulip/zulip/assets/56781761/22ac2aa8-bcbb-47ec-8739-6fca5145bf97) | ![Screenshot from 2023-10-09 00-51-44](https://github.com/zulip/zulip/assets/56781761/882a78a7-9506-4344-9f4f-b02f7a0edeef) |
| ![Screenshot from 2023-10-09 00-53-33](https://github.com/zulip/zulip/assets/56781761/75d21557-ca6d-4a94-aa50-1c6ce47f912d) | ![Screenshot from 2023-10-09 00-52-57](https://github.com/zulip/zulip/assets/56781761/db341d06-ebb3-4921-a2c0-1af2d613c46e) |



<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

prakhar1144 commented:
@terpimost your feedback would be helpful.

terpimost commented:
Border shouldn't be 2px:
<img width="1073" alt="image" src="https://github.com/zulip/zulip/assets/1903309/b31728fc-8c46-4fff-95de-863971df075f">
space between the edge of tabs rect and border should be 1 px:
<img width="617" alt="image" src="https://github.com/zulip/zulip/assets/1903309/131c4f69-484d-44d1-a156-b71cff6ead26">

prakhar1144 commented:
<details><summary>Light Theme</summary>
<img src="https://github.com/zulip/zulip/assets/56781761/1919accc-69ef-4031-923e-f3fd3653796b">
</img>
</details> 

<details><summary>Dark Theme</summary>
<img src="https://github.com/zulip/zulip/assets/56781761/1daf10c8-129d-4ddd-ae09-a5b2d5b5638d">
</img>
</details>

terpimost commented:
The hover state shouldn't be white background in light mode but 60% transparent white:
<img width="1341" alt="image" src="https://github.com/zulip/zulip/assets/1903309/84984768-1c9d-4ab5-a912-dc05a7397bc9">
On active state please do show the border: 
<img width="947" alt="image" src="https://github.com/zulip/zulip/assets/1903309/de4ad416-07b3-47f9-9d73-aa39cf86f7cc">

Could you please try to update follow icon? I think this version is properly aligned:
[icon-following](https://github.com/zulip/zulip/assets/1903309/0f76344b-44d3-4543-adfa-e96257d0566a)

prakhar1144 commented:
Light Theme

[vid1.webm](https://github.com/zulip/zulip/assets/56781761/b9389f0d-4724-47ab-b247-c07ff775eab2)

prakhar1144 commented:
Dark Theme

[vid2.webm](https://github.com/zulip/zulip/assets/56781761/dea83442-32bd-47e7-9cfa-355c232cb122)

terpimost commented:
Instead of having height 90% we should do `height: calc(100% - 4px);` in `.tabs-container .tab-option`
Icons are vertically not in the center of tabs, this is because of a strange rule here: 
<img width="1468" alt="image" src="https://github.com/zulip/zulip/assets/1903309/8bc6007c-a343-482b-bab6-c4f76f29fc52">

I think that rule was for the lists, so it would be correct to cancel it with some selector for this case.

I suggest also to correct left side bar mention and following icon style to be 40% transparent black in light theme and  45% transparent white in dark theme.

karlstolley commented on web/styles/app_components.css:
```diff
@@ -1221,14 +1222,18 @@ div.overlay {
-        height: 100%;
-        margin: 1px;
+        height: 90%;
+        margin: 1.5px;
```
Is there a reason to do half-pixel margins here? Those can't actually be rendered, but rounded up or down depending on the browser.

prakhar1144 replied:
Oh, wasn't sure that it get rounded. But makes sense.

We're no longer using margin in the updated code.

karlstolley commented on web/styles/app_components.css:
```diff
@@ -1221,14 +1222,18 @@ div.overlay {
+
+        &:active:not(.selected-tab) {
             outline: 1px solid hsl(0deg 0% 0% / 30%);
+            outline-offset: -1px;
```
I don't understand why this is an outline; an outline with offset -1 is basically a border, and the `border-box` sizing in this area should mean that setting a humble border here would work without incident.

prakhar1144 replied:
If we use border; then on action (mouse click) we add border which creates a wiggle effect.

karlstolley replied:
Ah okay. I'd usually just drop padding by a pixel on a hover/active effect that adds a border, but this works, too.

karlstolley commented on web/styles/app_components.css:
```diff
@@ -1221,14 +1222,18 @@ div.overlay {
 
         &:hover:not(.selected-tab) {
-            background-color: hsl(0deg 0% 100%);
+            background-color: hsl(0deg 0% 100% / 60%);
```
This PR presents a good opportunity to set CSS variables for the color values around this area, and fits with the "improve the CSS" theme of the PR, too.

prakhar1144 replied:
Added a prep commit

karlstolley commented on web/styles/app_components.css:
```diff
@@ -1221,14 +1222,18 @@ div.overlay {
         width: 100%;
-        height: 100%;
-        margin: 1px;
+        height: 90%;
```
I don't understand this `90%` height value. The container is 33px high, so why not just say that the inner tab is 31px high; that coupled with flexbox center properties will leave a 1px space around the buttons without having to specify a margin. Then, instead of left and right margins, just specify a `1px` gap on the `.tabs-container`.

prakhar1144 replied:
Used 31px for height.
1px padding on `.tabs-container` and 1px gap

I think we should keep the gap 2px as the gap is hardly visibile on 100% zoom ?

karlstolley replied:
Makes sense to me.

karlstolley commented:
Thanks for your work on this, @prakhar1144! I've left some comments and queries--happy to discuss further, as always.

prakhar1144 commented:
Addressed the review comments.

> I suggest also to correct left side bar mention and following icon style to be 40% transparent black in light theme and 45% transparent white in dark theme.

Started a [discussion on CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/left.20sidebar.20mention.20and.20follow.20icon/near/1663687)

timabbott commented:
This lgtm. I'm going to merge this as a useful checkpoint even if we have a few more design tweaks to do on top of this -- it feels like a clear improvement and I think the remaining changes are better tracked separately.
